readthedocs/projects/tasks.py

```diff
@@ -169,12 +170,17 @@ def run(self, pk, version_pk=None, build_pk=None, record=True, docker=False,
                                              build_env=self.build_env,
                                              config=self.config)
 
-            self.setup_environment()
+            try:
+                self.setup_environment()
 
-            # TODO the build object should have an idea of these states, extend
-            # the model to include an idea of these outcomes
-            outcomes = self.build_docs()
-            build_id = self.build.get('id')
+                # TODO the build object should have an idea of these states, extend
+                # the model to include an idea of these outcomes
+                outcomes = self.build_docs()
+                build_id = self.build.get('id')
+            except SoftTimeLimitExceeded:
+                raise BuildEnvironmentError(
+                    'Failing build because the maximum time to build the '
+                    'documentation was reached.')
```

Comment:
This should match the text from our docker container timeout:
https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/doc_builder/environments.py#L587-L588